directory/directory_test.go

```diff
@@ -144,12 +147,19 @@ func TestGetPutSignatures(t *testing.T) {
 		[]byte("sig1"),
 		[]byte("sig2"),
 	}
+	md, err := manifest.Digest(man)
+	require.NoError(t, err)
+
 	err = dest.SupportsSignatures()
 	assert.NoError(t, err)
-	err = dest.PutManifest(man)
+	err = dest.PutManifest(man, nil)
+	require.NoError(t, err)
+	err = dest.PutManifest(man, &md)
 	require.NoError(t, err)
 
-	err = dest.PutSignatures(signatures)
+	err = dest.PutSignatures(signatures, nil)
+	assert.NoError(t, err)
+	err = dest.PutSignatures(signatures, &md)
```

Comment:
Similarly, using two different sets of signatures would be better.

Reply:
Agreed, fixed.